vscode-extension/src/extension.ts

```diff
@@ -91,5 +92,12 @@ export async function activate(context: ExtensionContext) {
     }
   });
 
+  client.onReady().then(() => {
+    const clearIndexCacheCommand = commands.registerCommand("scalameta.clearIndexCache", async () => {
```

Comment:
Is this needed for every new command? I thought codeActions was a way for the server to list the commands it supports.

Reply:
Yes, it is. codeAction is not very well integrated in the client library, as of now. I guess it may depend by the requirement for extensions to statically include commands in the manifest